### ISSUE TITLE
Deploy Smart Contract using Foundry Fails with Deserialization Error: invalid length 1...

### ISSUE THREAD

**Issue by a-ridley** [hashgraph/hedera-json-rpc-relay#1824]
### Description
Back in January 2023, I successfully used `forge create` to deploy a smart contract on Hedera.
See closed issue #834 
Fast forward to October 2023 and I am running to the below error.

### What I did:
 When i run the following command
`forge create --rpc-url <your_rpc_url> --private-key <your_private_key> contracts/TodoList.sol:TodoList`
I received the following error:

```
Error: 
Deserialization Error: invalid length 1, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 1266. Response: {"timestamp":"0x65299eb0","difficulty":"0x0","extraData":"0x","gasLimit":"0xe4e1c0","baseFeePerGas":"0x1aebc97c400","gasUsed":"0x0","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","miner":"0x0000000000000000000000000000000000000000","mixHash":"0x0000000000000000000000000000000000000000000000000000000000000000","nonce":"0x0000000000000000","receiptsRoot":"0x0000000000000000000000000000000000000000000000000000000000000000","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","size":"0x59a","stateRoot":"0x0000000000000000000000000000000000000000000000000000000000000000","totalDifficulty":"0x0","transactions":[],"transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","uncles":[],"withdrawals":[],"withdrawalsRoot":"0x0","number":"0x320a0f","hash":"0xb734b7645ec3a5f3e5d51a15b4da8235f56486994c13f6b213be4eaa4b779cc4","parentHash":"0x563b39bf894273da3cd678b9c2952abb3561bb42ecf7750df127f47e3702ba84"}
```

![image](https://github.com/hashgraph/hedera-json-rpc-relay/assets/18725916/d323821c-eb33-4561-b6ee-a08f80f63dab)

### What I expected to happen:
My Contract to be deployed successfully and a similar message to the one below:

```
[⠰] Compiling...
No files changed, compilation skipped
Deployer: 0x4DdC478A1F35Bbf266D2F5767AE399f4d6E83857
Deployed to: 0x47c81E6e3f2880a42149574f08DCaBffCa10e164
Transaction hash: 0xfec3c5a6d726a4f94a57011c7c5f6fe3f54dad70700652570cda626f58a81dca
```

### Steps to reproduce

1. Unzip the project below and open it in vscode
[hedera_foundry_fork_testVendingMachine.zip](https://github.com/hashgraph/hedera-json-rpc-relay/files/10494681/hedera_foundry_fork_testVendingMachine.zip)
2. edit the .env file
3. Open a terminal in the project's dir
4. run `forge create --rpc-url https://testnet.hashio.io/api --private-key <HEX_Encoded_Private_Key> contracts/TodoList.sol:TodoList`

### Hedera network

testnet

### Version

latest

### Operating system

macOS

**Comment by Neurone** [comment]
This should be already fixed here: #1797

**Comment by Neurone** [comment]
In addition, while using foundry for smart contract development, because it uses ethers-rs, you will also likely face this problem: #1800 at least until we reintroduce the `v` parameter in the receipts.

**Comment by AlfredoG87** [comment]
Hello @Neurone and @a-ridley ,

I started to triage this issue last friday and was able to figure out the format issue with the `withdrawalsRoot` field that should be fixed after #1797  but does not seem to be fixed,
